lib/src/shared/main/java/com/couchbase/lite/AbstractReplicator.java

```diff
@@ -392,52 +395,56 @@ public void stop() {
     public Status getStatus() { return status.copy(); }
 
     @NonNull
-    public Set<String> getPendingDocumentIds() throws CouchbaseLiteException {
+    Set<String> getPendingDocumentIds() throws CouchbaseLiteException {
         if (config.getReplicatorType().equals(ReplicatorConfiguration.ReplicatorType.PULL)) {
             throw new CouchbaseLiteException(
                 "Pending Document IDs are not supported on pull-only replicators.",
-                Log.LOGGING_DOMAINS_TO_C4.get(DOMAIN),
+                CBLError.Domain.CBLITE,
                 CBLError.Code.UNSUPPORTED);
         }
 
         final C4Replicator repl;
-        synchronized (lock) { repl = c4repl; }
+        synchronized (lock) { repl = c4Replicator; }
 
-        // !!! Need to return a result here: will require recreating the replicator.
+        // ??? Not yet implemented
         if (repl == null) {
             Log.i(DOMAIN, "%s: Call to getPendingDocumentIds with unstarted replicator", this);
             return Collections.emptySet();
         }
 
         final Set<String> pending;
-        try { pending = c4repl.getPendingDocIDs(); }
+        try { pending = repl.getPendingDocIDs(); }
```

Comment:
this might need to be locked, else the same scenario Pasin mentioned can happen. repl can be stopped/offline/cleared from other thread.

Reply:
Yeah.  There was a bug there.  It isn't there now.  `repl` is a local ref and is not null.  There is no way that it can turn into null.  The C4 replicator to which it refers might be freed... but that is handled by `C4Replicator.getPendingDocIds`. There is no bug here.